### DISCUSSION TITLE
Consider implementing a `DeployDeterministicProtocol` script

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/v2-periphery#93] [General]
Ref https://github.com/sablier-labs/v2-periphery/pull/91#issuecomment-1580391265.

**Comment by smol-ninja** [comment]
Do we still need `DeployDeterministicProtocol` script? Because I remember we had this conclusion that precompiles cannot be used to deploy v2-core due to its large size of the bytecode.

**Reply by PaulRBerg** [reply]
I think we still want to implement it, given that we will merge the core and the periphery (package tethering).

AFAIK, the issue before was that the precompiles weren't exposed in the npm package. But now they are.

**Reply by smol-ninja** [reply]
We discussed it in this [issue](https://github.com/sablier-labs/v2-periphery/issues/238#issuecomment-1995357424), that a single transaction cannot deploy the entire protocol due to EVM gas limit. We tried to implement it after moving precompiles to the npm package [in here](https://github.com/sablier-labs/v2-periphery/pull/302#issue-2177251858).

**Reply by PaulRBerg** [reply]
Oooh, yes, I remember now. In this case, I will close this discussion.

This has implications for package tethering.
